contracts/reflect/src/testing.rs

```diff
@@ -1,53 +1,26 @@
 use crate::msg::{CustomQuery, CustomResponse};
 
-use cosmwasm_std::testing::{MockApi, MockQuerier, MockStorage};
-use cosmwasm_std::{
-    from_slice, to_binary, Binary, Coin, Extern, Querier, QuerierResult, QueryRequest, StdResult,
-    SystemError,
-};
+use cosmwasm_std::testing::{MockApi, MockQuerier, MockStorage, MOCK_CONTRACT_ADDR};
+use cosmwasm_std::{to_binary, Binary, Coin, Extern, HumanAddr, StdResult};
 
-#[derive(Clone)]
-pub struct CustomQuerier {
-    base: MockQuerier,
-}
-
-impl CustomQuerier {
-    pub fn new(base: MockQuerier) -> Self {
-        CustomQuerier { base }
-    }
-}
-
-impl Querier for CustomQuerier {
-    fn raw_query(&self, bin_request: &[u8]) -> QuerierResult {
-        // parse into our custom query class
-        let request: QueryRequest<CustomQuery> = match from_slice(bin_request) {
-            Ok(v) => v,
-            Err(e) => {
-                return Err(SystemError::InvalidRequest {
-                    error: format!("Parsing query request: {}", e),
-                    request: bin_request.into(),
-                })
-            }
-        };
-        if let QueryRequest::Custom(custom_query) = &request {
-            Ok(execute(&custom_query))
-        } else {
-            self.base.handle_query(&request)
-        }
-    }
-}
-
-/// mock_dependencies is a drop-in replacement for cosmwasm_std::testing::mock_dependencies
+/// A drop-in replacement for cosmwasm_std::testing::mock_dependencies
 /// this uses our CustomQuerier.
-pub fn mock_dependencies(
+pub fn mock_dependencies_with_custom_querier(
     canonical_length: usize,
     contract_balance: &[Coin],
-) -> Extern<MockStorage, MockApi, CustomQuerier> {
-    let base = cosmwasm_std::testing::mock_dependencies(canonical_length, contract_balance);
-    base.change_querier(CustomQuerier::new)
+) -> Extern<MockStorage, MockApi, MockQuerier<CustomQuery>> {
+    let contract_addr = HumanAddr::from(MOCK_CONTRACT_ADDR);
+    let custom_querier: MockQuerier<CustomQuery> =
+        MockQuerier::new(&[(&contract_addr, contract_balance)])
+            .with_custom_handler(|query| Ok(custom_query_execute(&query)));
```

Comment:
Why can't we replace this with:

`.with_custom_handler(custom_query_execute);` 

and just change the signature of `custom_query_execute`

Reply:
We can and in the integration test

```rust
    let contract_addr = HumanAddr::from(MOCK_CONTRACT_ADDR);
    let custom_querier: MockQuerier<CustomQuery> =
        MockQuerier::new(&[(&contract_addr, contract_balance)]).with_custom_handler(execute);

    Extern {
        storage: MockStorage::default(),
        api: MockApi::new(canonical_length),
        querier: custom_querier,
    }
```

it is shown that and how it works. I left it here this way to (a) have a demo using closures instead of function pointers and (b) keep the `custom_query_execute` interface as simple as possible because its logic is tested individually by unit tests.